test/broadcasting.jl

```diff
@@ -1490,14 +1490,18 @@ end
     @test @views typeof(df[!, 1:2]) <: SubDataFrame
 end
 
-@testset "broadcasting of df[:, col] = vector" begin
+@testset "broadcasting of df[:, col] = value" begin
     df = DataFrame(ones(3, 4))
     z = ["a", "b", "c"]
     df[:, :z] .= z
     @test df.z == z
     @test df.z !== z
     @test_throws ArgumentError df[:, 6] .= z
     @test_throws MethodError df[:, 1] .= z
+
+    df = DataFrame(ones(3, 4))
+    z = fill("abc", 1, 1, 1)
+    @test_throws DimensionMismatch df[:, :z] .= z
```

Comment:
This still doesn't test scalars, right? :-)

Reply:
right - I have accidentally removed the right test (this was an additional test I wanted to add, but it seems I have overwritten the correct one with this)